packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -22,6 +35,10 @@ abstract class ClusterParameterGroupBase extends Resource implements IClusterPar
    * The name of the parameter group
    */
   public abstract readonly clusterParameterGroupName: string;
+
+  public addParameter(_name: string, _value: string): AddParameterResult {
+    return { parameterAddedResult: AddParameterResultStatus.IMPORTED_RESOURCE_FAILURE };
```

Comment:
I can't see why you'd want a function that just returns a failure. I'm guessing that this got added because you can't actually call this function in this context (and the other place it exists) but that means you have the wrong level of abstraction here.

Reply:
I agree. The add operation should not default to failure. Failure should be the exception.